heapster-saw/doc/tutorial/tutorial.md

```diff
@@ -1057,12 +1045,10 @@ Definition zero_array_precond x
 
 
 We will not go into detail about the proof, but notice that the
-important steps are handled by custom automation. Namely we use the
-commands `prove_refinement_match_letRecM_l` and `prove_refinement` to
-handle refinements with and without recursion (respectively). The rest
-of the proof, consists of discharging simple inequalities and a couple
-of trivial entailments, where the left hand side is an error (and thus
-entails anything).
+important steps are handled by custom automation. 
+
+TODO: This proof is involved. Perhaps add some more details?
```

Comment:
Yes, I definitely think so! Are you planning to resolve this TODO in this PR, or are you leaving this as future work?

Reply:
It's a good point. The description more or less gets to the limit of what I understand, without going into technical details. Do you have any suggestion of how to improve this section?

We can also just commit the comment and leave it super high level at this point.

Reply:
Ideally, we would include a more detailed tutorial that goes into the nuances of `spec_refines` proofs involving loops. But writing such a tutorial would take a fair bit of effort, so I can understand if you don't want to do that here. My recommendation would be to:

1. Remove this TODO (since TODOs in a tutorial look out of place), and
2. Open an issue as a reminder to write this more detailed tutorial.